### DISCUSSION TITLE
No compatible package found for 'en_trf_bertbaseuncased_lg' (spaCy v3.0.0rc5)

### DISCUSSION THREAD

**Discussion by shrirang3010** [explosion/spaCy#8243] [Help: Installation] [answered]
## Your Environment

- - **spaCy version:** 3.0.0rc5
- **Platform:** Windows-10-10.0.19041-SP0
- **Python version:** 3.8.9
- **Pipelines:** en_core_web_sm (3.0.0)

**Comment by polm** [comment, marked as answer]
It sounds like you're using a v2 model name with v3. With v3, the transformer model has been renamed to `en_core_web_trf`, so you can install it like this:

    spacy download en_core_web_trf

See also [the install guide](https://spacy.io/usage). 

Also, we strongly recommend you do not use the rc release any more. There have been a number of bugfixes and minor improvements since then, and an rc is not intended for long term use anyway. We suggest you upgrade to the latest spaCy version.

**Reply by vasanthsarathy** [reply]
en_core_web_trf seems to be a ROBERTa model. How can I load other pretrained models like BERT (i.e., berbaseuncased, berbasecased etc.) and DISTILBERT in v3?

**Reply by adrianeboyd** [reply]
Please see the docs here: https://spacy.io/usage/embeddings-transformers#transformers-training

**Reply by vasanthsarathy** [reply]
Thank you. :) But, maybe I'm missing something, but don't I need to *download* the model? The link you pointed me to is for training the transformer model. Can I download another pretrained BERT model? All I want to be able to do at this time is to fire up the pretrained BERT, provide some input and run the "align" method on its data.

**Reply by adrianeboyd** [reply]
Ah, I see what you're asking. This is actually a bit frustrating to do because `init config` and `init fill-config` try to be smart about removing unused tok2vec/transformer components. So we're going to use `init config` to set up a config with a tagger and then remove it by hand:

```bash
spacy init config -p tagger -G trf.cfg
```

Edit `trf.cfg` to specify the transformer you want under `[components.transformer.model.name]` and **remove** the `tagger` from `nlp.pipeline` and delete all the `[components.tagger]` blocks.

Then use `spacy assemble` to create a model from the config, which is where the download happens:

```bash
spacy assemble trf.cfg /path/to/trf_model
```

```python
nlp = spacy.load("/path/to/trf_model")
```

Once you've done this once, you can just copy the config to assemble other transformer models. 

If you want, you can also use `spacy package` to package the model directory and then install it like other models so you can load from the package name instead of the path, just like `en_core_web_trf`.

You can also do this programmatically instead, which doesn't have the `init config` issues with removing components:

```python
import spacy
nlp = spacy.blank("en")
nlp.add_pipe("transformer", config={"model": {"name": "distilbert-base-uncased"}})
nlp.initialize()
nlp.to_disk("/path/to/trf_model")
```

However, a big warning: if you want to train additional components that use the transformer component, then you should use `init config`, because it configures a lot of training options for transformers correctly (batching, learning rate, etc.) that the programmatic option doesn't do, because its defaults are set up for non-transformer CPU training.

**Reply by Akshay0799** [reply]
@vasanthsarathy  you can probably do something like this and then call ner() on the data you want to run it on.
`import spacy 
from transformers import * 
ner = pipeline("ner", model="dslim/bert-base-NER")`

But I'm not sure if you can call 'align' on this, if you manage to find a workaround and still make it work, do let me know
